tests/test_precompiled.py

```diff
@@ -0,0 +1,459 @@
+import base64
+import io
+import json
+import uuid
+from io import BytesIO
+from unittest.mock import MagicMock
+
+import PyPDF2
+import pytest
+from flask import url_for
+from reportlab.lib.colors import white, black, grey
+from reportlab.lib.pagesizes import A4
+from reportlab.lib.units import mm
+from reportlab.pdfgen import canvas
+from reportlab.pdfgen.canvas import Canvas
+
+from app.precompiled import add_notify_tag_to_letter, validate_document
+from tests.pdf_consts import multi_page_pdf, not_pdf, blank_page, one_page_pdf, no_colour
+
+
+def test_precompiled_validation_endpoint_blank_pdf(client, auth_header):
+
+    response = client.post(
+        url_for('precompiled_blueprint.validate_pdf_document'),
+        data=blank_page,
+        headers={
+            'Content-type': 'application/json',
+            **auth_header
+        }
+    )
+
+    assert response.status_code == 200
+    json_data = json.loads(response.get_data())
+    assert json_data['result'] is True
```

Comment:
small thing - since flask 1.0 you can now call `response.json` rather than having json.loads